scipy/spatial/ckdtree/src/build.cxx

```diff
@@ -112,53 +120,50 @@ build(ckdtree *self, ckdtree_intp_t start_idx, intptr_t end_idx,
             auto mid = node_indices + n_points / 2;
             std::nth_element(
                 node_indices, mid, node_indices + n_points, index_compare);
-
             split = data[*mid * m + d];
-            p = partition_pivot(node_indices, mid, split);
+            p = partition_pivot(indices + start_idx, indices + end_idx, split);
```

Comment:
Why change this? We know the range `[node_indices, mid)` is partitioned by `nth_element` such that all values less than the pivot are in that range already.